app/assets/javascripts/forms.js

```diff
@@ -1,38 +1,107 @@
+// This file handles the behaviour for managing form building.
+//
+// Dependencies:
+//
+//   jQuery.ui.sortable: https://jqueryui.com/sortable/
+//   - Allows form fields to be dragged and re-oredered
+//
+//   Twitter's typeahead: http://twitter.github.io/typeahead.js/
+//   - Use for autocompleting for setting the field's name value
+
 (function(){
   var initialize = function(){
     makeSortable();
     bindHandlers();
+    autoComplete()
+  };
+
+  var whitelist = [
```

Comment:
This is a great example of some javascript that I think could really benefit from the structure of being a backbone view.